@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -443,6 +450,31 @@ function placePointLabelChoosingAnchor(
             outScreenPosition
         );
 
+        // Will be processed only once if the icon is rejected.
+        if (isRejected) {
```

Comment:
With your current code we always have isRejected === false in this function, cause rejected labels are handled already in placePointLabel.

Reply:
Of course I know about that, but I would strongly recommend to leave this function compatible with other `placePointLabelAtCurrentAnchor` function. It may be used in some other scenarios (lineMarkers, etc...) too.

Reply:
If you really insist and make my PR green (approved) then I will remove it :).

Reply:
yes, I insist. I agree with you that handling the rejected state is confusing the further you get from TextElementRenderer. So, if we can avoid those 20 lines full of ifs, much better.
The way you handle it at the beginning of placePointLabelChoosingAnchor is clean and simple.

Reply:
At this stage i would even rather remove isRejected from `placePointLabelAtCurrentAnchor ` as well, and just return Rejected for persistent labels with rejected icons at placePointLabel.

Reply:
Done.